controllers/glance_controller.go

```diff
@@ -822,7 +824,38 @@ func (r *GlanceReconciler) ensureRegisteredLimits(
 			ResourceName: lName,
 			DefaultLimit: lValue,
 		}
-		_, err = o.CreateRegisteredLimit(r.Log, m)
+		_, err = o.CreateOrUpdateRegisteredLimit(r.Log, m)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// registeredLimitsDelete - cleanup registered limits in keystone
+func (r *GlanceReconciler) registeredLimitsDelete(
+	ctx context.Context,
+	h *helper.Helper,
+	instance *glancev1.Glance,
+	quota map[string]int,
+) error {
+
+	// get admin
+	var err error
+	keystoneAPI, err := keystonev1.GetKeystoneAPI(ctx, h, instance.Namespace, map[string]string{})
+
+	if err != nil {
+		return err
+	}
+	o, _, err := glancev1.GetAdminServiceClient(ctx, h, keystoneAPI)
+	if err != nil {
+		return err
+	}
+	//TODO: This doesn't work properly atm as we have to wrap it further because
```

Comment:
@konan-abhi this is a WIP here: I still have to do some work to properly wrap the deletion into something that, starting from the `resourceName` makes the mapping to the `resourceID` and call the Delete function as per L858.

Reply:
I think here you can list all the resources based on serviceid (ListRegisteredLimitsByServiceID) or get particular resource using (ListRegisteredLimitsByResourceName) method and then call the further deletion method over it.